src/scene.ts

```diff
@@ -3993,6 +3993,8 @@ export class Scene extends AbstractScene implements IAnimatable, IClipPlanesHold
             return;
         }
 
+        this.onBeforeCameraRenderObservable.notifyObservers(camera);

```

Comment:
I don't think you can move the notification, it is a breaking change.

Current notified code expects that `_activeCamera`, viewport, `_renderId`, etc are setup when they are called. By moving the notification here it does not stand anymore.

Reply:
Though I totally agree with this change, this is in fact a breaking change. I have tried fixing that in an earlier PR (can't find the reference right now, sorry), but it has side effects that will influence different scenes. The main issue when I tested it was the setViewport, that until now was expected to be called before the observable is triggered, and in this case is executed after.

Reply:
Rolled back these changes.